blocks/secure-assets/secure-assets.js

```diff
@@ -0,0 +1,152 @@
+// eslint-disable-next-line import/no-unresolved
+import { SignJWT } from 'https://cdnjs.cloudflare.com/ajax/libs/jose/5.2.3/jwt/sign.js';
+
+const securedImages = [];
+let placeholderImg;
+
+/**
+ * to check if given src is a DM OpenAPI URL
+ */
+function isDMOpenAPIUrl(src) {
+  return /^(https?:\/\/(.*)\/adobe\/assets\/urn:aaid:aem:(.*))/gm.test(src);
+}
+
+/**
+ * to check if given src is a secure asset
+ */
+async function isSecureAsset(src) {
+  let isSecure = false;
+  try {
+    const response = await fetch(src, { method: 'HEAD' });
+    if (response.ok) {
+      isSecure = false;
+    } else if (response.status === 404) {
+      // 404 is thrown by DM OpenAPI if the asset is secure
+      isSecure = true;
+    }
+  } catch {
+    isSecure = false;
+  }
+  return isSecure;
+}
+
+async function reloadImage(img, token) {
+  await fetch(img.getAttribute('data-src'), {
+    headers: {
+      'x-asset-delivery-token': token,
+    },
+  }).then(async (resp) => {
+    if (resp.status === 200) {
+      const reader = new FileReader();
+      reader.onload = () => {
+        const base64Data = reader.result;
```

Comment:
Why do we need to fetch and convert to base64? Would just suffixing the token (once available) to the delivery url suffice

Reply:
I'm uncertain how to prompt an <img> tag to embed a delivery token while calling a secure asset URL. When I reattach the secure asset URL to the tag attributes, it throws a 404 error due to the missing delivery token. Therefore, the approach involves manually calling the URL response by embedding the delivery token, converting the response into base64, and embedding it in the <img> tag attributes.

Reply:
ok yes you need to send the header in the request and for that you need to make the fetch request programatically.